packages/core/test/integration/json.test.ts

```diff
@@ -213,13 +213,21 @@ describe('JSON Querying', () => {
         }],
       });
 
-      // we can't automatically detect that the output is JSON type in mariadb < 10.4.3,
+      // we can't automatically detect that the output is JSON type in mariadb < 10.5.2,
       // and we don't yet support specifying (nor inferring) the type of custom attributes,
       // so for now the output is different in this specific case
-      const expectedResult = dialectName === 'mariadb' && semver.lt(sequelize.getDatabaseVersion(), '10.4.3') ? '"swen"' : 'swen';
-
-      // @ts-expect-error -- getDataValue does not support custom attributes
-      expect(orders[0].user.getDataValue('name')).to.equal(expectedResult);
+      if (dialectName === 'mariadb' && semver.lt(sequelize.getDatabaseVersion(), '10.5.2')) {
+        if (semver.gt(sequelize.getDatabaseVersion(), '10.4.21')) {
+          // @ts-expect-error -- getDataValue does not support custom attributes
+          expect(orders[0].user.getDataValue('name')).to.equal('"swen"');
+        }
+
+        // @ts-expect-error -- getDataValue does not support custom attributes
+        expect(orders[0].user.getDataValue('name')).to.deep.equal(Buffer.from('"swen"'));
```

Comment:
I don't really know where this comes from, but I'm not sure if we want to fix it or increase the minimum patch version to 10.4.22. @ephys any preference?

Release notes of 10.4.22 are here; https://mariadb.com/kb/en/mariadb-10422-release-notes/ with a link to the changelog

Reply:
These changes are very weird, so I'll wait to see if changing the version to 10.4.30 helps with this